backend/history-service/models/history.model.js

```diff
@@ -0,0 +1,16 @@
+const mongoose = require('mongoose');
+
+const historySchema = new mongoose.Schema({
+    questionId: String,
+    language: String,
+    solution: String,
+    userId: String,
+    authors: [
+        {
+            user_id: String,
```

Comment:
Could I ask for clarification on the necessity of both `userId` and `user_id`?

Reply:
The original implementation only had authors but since there weren't necessarily 2 authors for every attempt, the index of the main author got confused so I just kept the main author's id as userId and authors as a place to store collaborators. Also, userId is easier to search if say I wanted to bring up a specific user's attempts in the history page compared to checking the authors' array that will have 2 authors where the user might be but he wasn't the one who saved the attempt. TLDR userId - user who saved the attempt and authors - collaborators on solution.

Reply:
I can delete the authors array since there will be at most 1 collaborator but that's just how the implementation is for now.

Reply:
@YZTangent thoughts?

Reply:
Since we're saving individually, the the authors arary is redundant, I'll go remove it. I wrote it with the intention of accounting for autosaving which saves the history to both users, but since we're not doing that then its not needed.